examples/src/main/java/jaxrs/examples/multipart/MultipartClient.java

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************
+* Copyright (c) 2021 Eclipse Foundation
+*
+* This specification document is made available under the terms
+* of the Eclipse Foundation Specification License v1.0, which is
+* available at https://www.eclipse.org/legal/efsl.php.
+*******************************************************************/
+package jaxrs.examples.multipart;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.client.Entity;
+import jakarta.ws.rs.client.WebTarget;
+import jakarta.ws.rs.core.GenericType;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.ext.Part;
+
+public class MultipartClient {
+
+    public boolean sendPdfs(File dir) throws IOException {
+        List<Part> parts = new ArrayList<>();
+        for (File f : dir.listFiles()) {
+            parts.add(Part.newBuilder(f.getName()).entityStream(f.getName(), new FileInputStream(f))
```

Comment:
Yeah, that's a good question - here are a few thoughts (assuming that we change `newBuilder(...)` to `withName(...)`:
(1) We could add a `withFileName(String filename)` method that would set both the name and filename to the passed in String value.
(2) We could add a `withName(String name, boolean sameAsFileName)` - and when `sameAsFileName` is true, it would be equivalent to calling `withName(name).fileName(name)`
(3) Some other convention?

I think I lean towards (1).

Reply:
Yes, definitely (1).